sources/engine/Stride.UI/UIDocumentExtensions.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) .NET Foundation and Contributors (https://dotnetfoundation.org/ & https://stride3d.net)
+// Distributed under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using Stride.Core.Mathematics;
+using Stride.Engine;
+using Stride.Graphics;
+using Stride.Rendering;
+
+namespace Stride.UI;
+
+public static class UIDocumentExtensions
```

Comment:
Because `UIRenderFeature` uses the exact same methods on the `RenderUIDocument` (they are almost 1 - 1 copy-pasted). So I was intending to inroduce a interface with the common properties, which `UIDocument`, `RenderUIDocument`, and even `UIComponent` could implement. But ended up feeling kind of messy the way I did it. I was going to either try again or move them to be part of the `UIDocument` class. But I guess I just plain forgot to do it before committing. Do you have a preferences of which way I go about it?

Reply:
An interface sounds fine, what did you have trouble with when implementing that solution ?